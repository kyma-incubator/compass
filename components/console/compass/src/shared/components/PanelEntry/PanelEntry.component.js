import { Panel } from 'fundamental-react';
import React from 'react';
import PropTypes from 'prop-types';

const PanelEntry = ({ title, children }) => (
  <Panel.Body>
    <p className="fd-has-color-text-4 fd-has-margin-bottom-none">{title}</p>
    {children}
  </Panel.Body>
);

PanelEntry.propTypes = {
  title: PropTypes.string.isRequired,
  children: PropTypes.node.isRequired,
};

export default PanelEntry;
