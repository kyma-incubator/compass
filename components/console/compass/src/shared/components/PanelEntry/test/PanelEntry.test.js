import React from 'react';
import renderer from 'react-test-renderer';
import PanelEntry from '../PanelEntry.component';

describe('PanelEntry', () => {
  it(`Renders title and children`, () => {
    const component = renderer.create(
      <PanelEntry title="testtitle" children="testcontent" />,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
