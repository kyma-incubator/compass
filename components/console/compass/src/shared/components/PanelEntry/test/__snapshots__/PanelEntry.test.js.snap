// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PanelEntry Renders title and children 1`] = `
<div
  className="fd-panel__body"
>
  <p
    className="fd-has-color-text-4 fd-has-margin-bottom-none"
  >
    testtitle
  </p>
  testcontent
</div>
`;
